core/lib/spree/app_configuration.rb

```diff
@@ -363,6 +363,14 @@ def payment_canceller
     #   Spree::CurrentStoreSelector
     class_name_attribute :current_store_selector_class, default: 'Spree::StoreSelector::ByServerName'
 
+    # Allows providing your own class for inflecting strings.
+    # For example, to create the permalinks for taxons.
+    #
+    # @!attribute [rw] string_inflection_class
+    # @return [Class] a class that provides `#parameterize`, and `#dasherize`
+    # methods that both return Strings
+    class_name_attribute :string_inflection_class, default: 'ActiveSupport::Inflector'
```

Comment:
This needs a name more clearly scoped to what it does. We (through rials) use `ActiveSupport::Inflector` heavily for other things and this name shouldn't imply that we are overriding that. It should be clear that this is for URL parameterization only.

The question here is do we intend to use this for. If we intend to only use this class from Taxon, it should be called `taxon_parametizer_class` (or maybe `taxon_url_parametizer_class`). If we intend to move everything else over, we could call it `parametizer_class` (or maybe `url_parametizer_class`).

It's worth noting that ![friendly_id](https://github.com/norman/friendly_id) has some of its own logic and configuration for parameterization (though I believe it relies on `Inflector.parameterize`). I'm not sure we would want to lose that logic, but it would be nice if we generated URLs consistently.